packages/validator/src/services/validatorStore.ts

```diff
@@ -191,18 +223,79 @@ export class ValidatorStore {
   }
 
   getGasLimit(pubkeyHex: PubkeyHex): number {
-    return (
-      (this.validators.get(pubkeyHex)?.builder || {}).gasLimit ??
-      this.defaultProposerConfig?.builder.gasLimit ??
-      defaultOptions.defaultGasLimit
-    );
+    if (this.validators.has(pubkeyHex)) {
+      return (
+        (this.validators.get(pubkeyHex)?.builder || {}).gasLimit ??
+        this.defaultProposerConfig?.builder.gasLimit ??
+        defaultOptions.defaultGasLimit
+      );
+    } else {
+      throw Error(`Validator pubkey ${pubkeyHex} not known`);
+    }
+  }
+
+  setGasLimit(pubkeyHex: PubkeyHex, gasLimitInput: string | number): void {
+    if ((typeof gasLimitInput !== "string" && typeof gasLimitInput !== "number") || `${gasLimitInput}`.trim() === "") {
+      throw Error("Not valid Gas Limit");
+    }
+    const gasLimit = Number(gasLimitInput);
+    if (Number.isNaN(gasLimit) || gasLimit === 0) {
+      throw Error(`Gas Limit is not valid gasLimit=${gasLimit}`);
+    }
```

Comment:
The ValidatorStore in not a place to parse unsanitized data. Please parse gasLimit into a usable typed value where it's collected from the user be it API or CLI